README.md

```diff
@@ -4,222 +4,1008 @@
+# Quick Start
 
-* [JCLI's test runner](https://github.com/SealabJaster/jcli/blob/master/examples/test.d)
+*This is a brief overview, for slightly more in-depth examples please look at the fully-documented [examples](https://github.com/BradleyChatha/jcli/tree/master/examples) folder.*
 
-* [JCLI's manual/misc test program](https://github.com/SealabJaster/jcli_testerr)
+## Creating a default command
 
-## Changelog
+The default command is the command that is ran when you don't specify the name of a named command. e.g. `mytool 60 20 --some=args` would call the default command if it exists:
 
-I publish changelogs over at [my website](https://bradley.chatha.dev/Blog/JcliNews).
+```d
+// inside of app.d
+module app;
+import jaster.cli;
 
-## Quick Start
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    int onExecute()
+    {
+        return 0;
+    }
+}
+```
 
-Include this library in your project (e.g. `dub add jcli`).
+The `@Command` is a UDA (User Defined Attribute) where the first parameter is the command's name (`null` for the default command), and the second parameter is the command's description.
 
-Create a class or struct; attach `@Command("commandname")` onto it; add as many public variables as you want, with `@CommandNamedArg` or `@CommandPositionalArg` attached; 
-add an `onExecute` function that returns either `void` or `int`:
+All commands must define an `onExecute` function, which either returns `void`, or an `int` that will be used as the program's exit/status code.
 
-```d
-module mytool.commands;
+As a side note, an initial dub project does not include the intial `module app;` shown in the example above. I've added it as we'll need the directly reference the module in a later section.
 
-import jaster.cli;
+## Positional Arguments
 
-// Matches: mytool compile all
-//          mytool compile
-//          mytool ct
-@Command("compile all|compile|ct")
-struct CompileCommand
-{
-    // Use `Nullable` for optional arguments.
-    // `Nullable` is publicly imported by JCLI for ease-of-use.
-    @CommandNamedArg("o|output", "Where to place the compiled output")
-    Nullable!string output;
+To start off, let's make our default command take a number as a positional arg. If this number is even then return `1`, otherwise return `0`.
 
-    @CommandPositionalArg(0, "The file to compile")
-    string toCompile;
+Positional arguments don't have a name, and are expected to exist in a specific position within the arguments passed to your program.
 
-    int onExecute()
-    {
-        import std.file : exists;
+For example the command `mytool 60 yoyo true` would have `60` in the 0th position, `yoyo` in the 1st position, and `true` in the 2nd position:
 
-        if(!exists(this.toCompile))
-            return -1; // Return an int if you want to control the exit code.
+```d
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    @CommandPositionalArg(0, "number", "The number to check.")
```

Comment:
Is it possible to inherit index from the order of `@CommandPositionalArg` members like below?
```d
@CommandPositionalArg("number", "The number to check.")
int number;

@CommandPositionalArg("text", "The number to check.")
string text;
```
So `number` has `0` index and `text` has `1`.

Reply:
It's completely possible.

However, having behaviour that is defined by lexical order is something I'd like to avoid. `ArgBinder` being an exception since I can't think of a decent way of handling its logic.

While there are arguments such as "How often are users going to reorder their members inside of the code?", I think it's for the best to just completely prevent that from happening in the first place, at the cost of manually incrementing a number for every positional arg.

Reply:
I guess so. Question now is simply what how to decide between the two types.

Maybe just `CommandAutoPositionalArg`